libs/core/src/service/search_service.rs

```diff
@@ -400,6 +400,20 @@ impl<Client: Requester, Docs: DocumentService> CoreState<Client, Docs> {
     }
 }
 
+impl SearchResult {
+    pub fn get_score(&self) -> Option<i64> {
+        match self {
+            SearchResult::FileNameMatch { id: _, path: _, matched_indices: _, score } => {
+                Some(*score)
+            }
+            SearchResult::FileContentMatches { id: _, path: _, content_matches } => {
+                Some(content_matches[0].score)
```

Comment:
Choosing the first item to represent the score of the entire content match list is going to give somewhat inaccurate results.

Reply:
Although you separated the file content matches, their scores are all still reliant on the first item's score. This will push up more irrelevant matches from the same file. You should make each file content match from the same file be sorted by its own score.